libs/ledger/src/storage_unit/transaction_store_sync_service.cpp

```diff
@@ -515,21 +516,13 @@ bool TransactionStoreSyncService::AlreadySeen(chain::Transaction const &tx)
     result = true;
   }
 
-  if (recently_seen_txs_.find(digest) == recently_seen_txs_.end())
+  if (recently_seen_txs_.Seen(digest))
   {
-    // TX not recently seen, add it and trim
-    recently_seen_txs_.insert(digest);
-    recently_seen_txs_ordered_.push_front(digest);
-
-    while (recently_seen_txs_.size() > RECENTLY_SEEN_CACHE_SIZE)
-    {
-      recently_seen_txs_.erase(recently_seen_txs_ordered_.back());
-      recently_seen_txs_ordered_.pop_back();
-    }
+    result = true;
   }
   else
   {
-    result = true;
+    recently_seen_txs_.Add(digest);
   }
```

Comment:
A microimprovement: this whole block logic can be hidden in `HaveSeenRecentlyCache::Add()` which can in turn return whether an insertion did actually happen, same as `std::(unordered_)?set::(insert|emplace)` do it.

Reply:
I did consider this, but, this makes it a bit ambigous to the reader. The logical thing to return from Add is 'did it add to the cache', which you would expect to be always true.